### PR TITLE
[full-ci] Search batch actions

### PR DESCRIPTION
**Pull request by JanAckermann**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by JanAckermann** [comment]
@lookacat this is so early in progress -> no changelog and most of the stuff is missing and not working

File: packages/web-app-files/src/composables/actions/helpers/useFileActionsDeleteResources.ts [hunk 1]
```diff
@@ -170,69 +177,90 @@ export const useFileActionsDeleteResources = ({ store }: { store?: Store<any> })
...
-              // reset pagination to avoid empty lists (happens when deleting all items on the last page)
-              currentPageQuery.value = pageCount.toString()
-            }
-          })
-      },
-      { indeterminate: false }
-    )
+    const resourceSpaceMapping: Record<string, { space: SpaceResource; resources: Resource[] }> = {}
+
+    for (const resource of unref(resources)) {
```
Comment (JanAckermann) [inline packages/web-app-files/src/composables/actions/helpers/useFileActionsDeleteResources.ts:182]: Use acc

File: packages/web-app-files/src/composables/actions/helpers/useFileActionsDeleteResources.ts [hunk 2]
```diff
@@ -170,69 +177,90 @@ export const useFileActionsDeleteResources = ({ store }: { store?: Store<any> })
   }
 
   const filesList_delete = (space: SpaceResource) => {
```
Comment (JanAckermann) [inline packages/web-app-files/src/composables/actions/helpers/useFileActionsDeleteResources.ts:179]: Check if space is still needed in fnc Signatur

File: packages/web-app-files/src/composables/actions/helpers/useFileActionsDeleteResources.ts [hunk 3]
```diff
@@ -170,69 +177,90 @@ export const useFileActionsDeleteResources = ({ store }: { store?: Store<any> })
...
+      if (!(matchingSpace.id in resourceSpaceMapping)) {
+        resourceSpaceMapping[matchingSpace.id] = { space: matchingSpace, resources: [] }
+      }
+
+      resourceSpaceMapping[matchingSpace.id].resources.push(resource)
+    }
+
+    const promises = []
+
+    for (const { space: spaceForDeletion, resources: resourcesForDeletion } of Object.values(
```
Comment (JanAckermann) [inline packages/web-app-files/src/composables/actions/helpers/useFileActionsDeleteResources.ts:194]: Use Lodash for

**Comment by JanAckermann** [comment]
Todo: move getMatchingSpace to own  composable.

**Comment by JanAckermann** [comment]
Blocked by https://github.com/owncloud/ocis/issues/6525 no permissions on resources in space -> canDelete results in false

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue [hunk 4]
```diff
@@ -799,14 +796,7 @@ export default defineComponent({
...
     },
     emitFileClick(resource) {
-      let space = this.resourceRouteResolver.getMatchingSpace(resource)
-      if (!space) {
-        space = buildShareSpaceResource({
-          shareId: resource.shareId,
-          shareName: resource.name,
-          serverUrl: configurationManager.serverUrl
-        })
-      }
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/FilesList/ResourceTable.vue:809]: this seems to be double, as already in getMatchingSpace

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS-2</strong> https://drone.owncloud.com/owncloud/web/36570/13/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36570/tracing/participant-management-alice-2023-6-15-09-59-27.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36570/tracing/participant-management-anonymous-2023-6-15-09-59-58.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36570/tracing/participant-management-brian-2023-6-15-09-59-36.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36570/tracing/participant-management-carol-2023-6-15-09-59-47.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36570/tracing/participant-management-david-2023-6-15-09-59-39.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/36570/tracing/participant-management-edith-2023-6-15-09-59-43.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/components/FilesList/KeyboardActions.vue [hunk 5]
```diff
@@ -33,6 +34,8 @@ export default defineComponent({
     const clientService = useClientService()
     const { scrollToResource } = useScrollTo()
     const loadingService = useLoadingService()
+    const { actions: pasteFileActions } = useFileActionsPaste({ store })
+    const pasteFileAction = computed(() => unref(pasteFileActions)[0].handler)
```
Comment (JanAckermann) [inline packages/web-app-files/src/components/FilesList/KeyboardActions.vue:38]: does this needs to be a computed ?

File: packages/web-app-files/src/composables/actions/files/useFileActionsPaste.ts [hunk 6]
```diff
@@ -27,17 +35,42 @@ export const useFileActionsPaste = ({ store }: { store?: Store<any> } = {}) => {
...
+              $gettext,
+              $gettextInterpolate,
+              $ngettext
+            })
+          },
+          { debounceTime: 0 }
+        )
+      }
+    )
+    return Promise.all(promises).then(() => {
```
Comment (JanAckermann) [inline packages/web-app-files/src/composables/actions/files/useFileActionsPaste.ts:72]: await

Comment (JanAckermann) [reply packages/web-app-files/src/composables/actions/files/useFileActionsPaste.ts:72]: enable conflict handling when 2 resources with same name from different spaces will be pasted

File: packages/web-app-files/src/composables/filesList/useResourceRouteResolver.ts [hunk 7]
```diff
@@ -2,39 +2,23 @@ import { unref, Ref } from 'vue'
...
 import { createLocationSpaces, createLocationShares } from '../../router'
 import { CreateTargetRouteOptions } from '../../helpers/folderLink/types'
-import { Resource, SpaceResource } from 'web-client/src'
+import { Resource } from 'web-client/src'
 import { buildShareSpaceResource } from 'web-client/src/helpers'
 
 type ResourceRouteResolverOptions = {
   configurationManager?: ConfigurationManager
   targetRouteCallback?: Ref<any>
-  space?: Ref<SpaceResource>
```
Comment (JanAckermann) [inline packages/web-app-files/src/composables/filesList/useResourceRouteResolver.ts:15]: introduce this again

File: packages/web-app-files/src/store/actions.ts [hunk 8]
```diff
@@ -99,39 +101,34 @@ export default {
       $ngettext,
       $gettextInterpolate
     )
-    let movedResources = []
+    let movedResourcesPromise
     if (context.state.clipboardAction === ClipboardActions.Cut) {
```
Comment (JanAckermann) [inline packages/web-app-files/src/store/actions.ts:105]: purge store action, move to composable

File: packages/web-client/src/helpers/space/functions.ts [hunk 9]
```diff
@@ -206,6 +206,9 @@ export function buildSpace(data): SpaceResource {
     isReceivedShare: function () {
       return false
     },
+    isShareRoot: function () {
+      return false
```
Comment (JanAckermann) [inline packages/web-client/src/helpers/space/functions.ts:210]: driveType mountpoint || share return true

**Comment by JanAckermann** [comment]
Scenario `OCIS=true  pnpm test:e2e:cucumber 'tests/e2e/cucumber/features/smoke/spaces/participantManagement.ocis.feature:3'`
is failing.

Basically, that's deleting a resource in a public share that's inside a project space

File: packages/web-client/src/helpers/space/functions.ts [hunk 10]
```diff
@@ -206,6 +206,9 @@ export function buildSpace(data): SpaceResource {
     isReceivedShare: function () {
       return false
     },
+    isShareRoot: function () {
+      return ['share', 'mountpoint'].includes(data.driveType)
```
Comment (JanAckermann) [inline packages/web-client/src/helpers/space/functions.ts:210]: what's with public?

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/36615/59/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesCopy-copy_feature-L76.png
  


</p></details>

File: packages/web-pkg/src/composables/spaces/useGetMatchingSpace.ts [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  const store = useStore()
+  const configurationManager = useConfigurationManager()
+  const spaces = store.getters['runtime/spaces/spaces']
+  const { $gettext } = useGettext()
+  const getInternalSpace = (storageId) => {
+    return spaces.find((space) => space.id === storageId)
+  }
+
+  const getMatchingSpace = (resource: Resource): SpaceResource => {
+    if (resource.downloadURL.split('/')?.[5] === 'public-files') {
```
Comment (JanAckermann) [inline packages/web-pkg/src/composables/spaces/useGetMatchingSpace.ts:22]: This is not acceptable :-1:

Comment (kulmann) [reply packages/web-pkg/src/composables/spaces/useGetMatchingSpace.ts]: should not be needed anymore with my latest commit. can you check/confirm?

Comment (kulmann) [reply packages/web-pkg/src/composables/spaces/useGetMatchingSpace.ts]: was still needed, but we made it simpler together :heart:

**Review by kulmann — APPROVED** [review]
👏 👏 👏

**Review by lookacat — APPROVED** [review]
Good work <3

**Pull request merged**